### PR TITLE
Fix issue-257 and Streamline parse_date_time

### PR DESCRIPTION
**Pull request by labrys**
Fixes sickrage/sickrage-issues#257
Catch networks that aren't in Network timezones and use sb_timezone instead.
Clean up regex for time parsing and rewrite parse_date_time to use named groups.

### CODE REVIEW

**Comment by SickRage2** [comment]
Thanks for the pull request! Before a real human comes by, please make sure your PR has all of the below criteria checked
- [x] Give a description on what the PR is for.
- [x] Make sure your PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review

Please make sure you also read [contribution guide](https://github.com/SickRage/SickRage/blob/master/contributing.md) and followed all of the steps.

Thanks!

**Comment by fernandog** [comment]
@labrys 
when network is added to network time zone, will the user get the updated info?

**Comment by labrys** [comment]
@fernandog on deleting cache.db, definitely, but I haven't looked into when we actually pull the data from github

File: sickbeard/network_timezones.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def update_network_dict():
 
     queries = []
     for network, timezone in d.iteritems():
-        existing = network_list.has_key(network)
+        existing = network in network_list
         if not existing:
```
Comment (miigotu) [inline sickbeard/network_timezones.py]: `if network not in network_list:` ?

**Pull request merged**